src/parser/index.js

```diff
@@ -390,6 +388,74 @@ export class Parser {
     return this.consume().value
   }
 
+  /**
+   * Try to parse a type list.
+   *
+   * @return {Array<string>} The list of types
+   */
+  typeList () {
+    if (!this.matches('COLON') &&
+      (this.peek().type !== 'TYPE' || this.peek().type !== 'LEFT_PAREN')) {
+      // TODO Better error
+      this.report(`Expected a type or a list of types, got "${this.peek().type}"`)
+    }
+
+    // We just have a single type
+    if (!this.matches('LEFT_PAREN')) {
+      return [{
+        type: this.consume().value,
+        selected: true
+      }]
+    }
+
+    let typeList = []
+    while (!this.eof() && !this.matches('RIGHT_PAREN')) {
+      // Check if the type is preceded by a bracket to denote
+      // that this is the type of the return value we want.
+      let selected = this.matches('LEFT_BRACKET')
+      if (!this.peek().type === 'TYPE') {
+        this.report(`Unexpected identifier in type list, expected type, got "${this.peek().type}"`)
+      }
+
+      typeList.push({
+        type: this.consume().value,
+        selected
+      })
+
+      // If the type was preceded by a left bracket, then it
+      // should be followed by a right bracket.
+      if (selected && !this.matches('RIGHT_BRACKET')) {
+        this.report(`Unclosed selected type`)
+      }
+
+      // Break if the next character is not a comma or a right parenthesis
+      // If this is true, then we are specifying more types without
+      // delimiting them using comma.
+      if (!this.matches('COMMA') &&
+        this.peek().type !== 'RIGHT_PAREN') break
```

Comment:
Wondering if we should report an error here too; if you did something like `foo(): (uint256 uint256)`, I think we should report a parsing error.

Reply:
That would be inconsistent with https://github.com/aragon/radspec/blob/master/src/parser/index.js#L399-L420

Reply:
> If this is true, then we are specifying more parameters without delimiting them using comma.

Hmm, I don't know if we really want this behaviour. What do you think about being strict with parameters requiring a comma to split them? Do you remember if there was an explicit reason to allow spaces to be the delimiter?.

Reply:
No, I don't remember.